Building using GCC 15 (C23) fails related to bools

Building rc v1.7.4 using GCC 15 (C23) on Fedora Rawhide fails like this:

```
gcc -DHAVE_CONFIG_H -I.     -Wall -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer  -c -o exec.o exec.c
make[1]: Leaving directory '/builddir/build/BUILD/rc-1.7.4-build/rc-1.7.4'
In file included from exec.c:3:
rc.h:47:14: error: expected ‘{’ before ‘bool’
   47 | typedef enum bool {
      |              ^~~~
rc.h:47:14: error: two or more data types in declaration specifiers
rc.h:49:1: warning: useless type name in empty declaration
   49 | } bool;
      | ^
exec.c: In function ‘exec’:
exec.c:23:34: error: ‘FALSE’ undeclared (first use in this function)
   23 |         saw_builtin = saw_exec = FALSE;
      |                                  ^~~~~
exec.c:23:34: note: each undeclared identifier is reported only once for each function it appears in
exec.c:40:36: error: ‘TRUE’ undeclared (first use in this function)
   40 |                         saw_exec = TRUE;
      |                                    ^~~~
make[1]: *** [Makefile:551: exec.o] Error 1
```

--- a/rc.h
+++ b/rc.h
@@ -47,9 +47,14 @@ typedef enum ecodes {
 	eError, eBreak, eReturn, eVarstack, eArena, eFifo, eFd, eContinue
 } ecodes;
 
+#if __STDC_VERSION__ >= 202311L
+#define FALSE false
+#define TRUE true
+#else
 typedef enum bool {
 	FALSE, TRUE
 } bool;
+#endif
 
 typedef enum redirtype {
 	rFrom, rCreate, rAppend, rHeredoc, rHerestring

